source/containers.tex

```diff
@@ -9200,6 +9238,665 @@
+
+\indexlibrarymember{try_append_range}{inplace_vector}%
+\begin{itemdecl}
+template<@\exposconcept{container-compatible-range}@<T> R>
+  constexpr ranges::borrowed_iterator_t<R> try_append_range(R&& rg);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{T} is \oldconcept{EmplaceConstructible}
+from \tcode{*ranges::begin(rg)} into \tcode{*this}.
+
+\pnum
+\effects
+Appends copies of initial elements in \tcode{rg} before \tcode{end()},
+until all elements are inserted or \tcode{size() == capacity()} is \tcode{true}.
+Each iterator in the range \tcode{rg} is dereferenced at most once.
+
+\pnum
+\returns
+%FIXME: Should this be "An iterator past the last inserted element of rg."?
+Iterator past last inserted element of \tcode{rg}.
+
+\pnum
+\complexity
+Linear in the number of elements inserted.
+
+\pnum
+\remarks
+Let $n$ be the value of \tcode{size()} prior to this call.
+If an exception is thrown after insertion of $k$ elements, then
+\tcode{size()} equals $n + k$,
+%FIXME: Are these valid ranges?
+elements in the range \tcode{begin() + \range{0}{$n$}} are not modified, and
+elements in the range \tcode{begin() + \range{$n$}{$n + k$}} correspond to
+the inserted elements.
+\end{itemdescr}
+
+\indexlibrarymember{unchecked_emplace_back}{inplace_vector}%
+\begin{itemdecl}
+template<class... Args>
+  constexpr reference unchecked_emplace_back(Args&&... x);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{size() < capacity()} is \tcode{true}.
+
+\pnum
+\effects
+Equivalent to:
+\tcode{return *try_emplace_back(std::forward<Args>(args)...);}
+\end{itemdescr}
+
+\indexlibrarymember{unchecked_push_back}{inplace_vector}%
+\begin{itemdecl}
+constexpr reference unchecked_push_back(const T& x);
+constexpr reference unchecked_push_back(T&& x);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{size() < capacity()} is \tcode{true}.
+
+\pnum
+\effects
+Equivalent to:
+\tcode{return *try_push_back(std::forward<decltype(x)>(x));}
+\end{itemdescr}
+
+\indexlibrarymember{reserve}{inplace_vector}%
+\begin{itemdecl}
+static constexpr void reserve(size_type n);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+None.
+
+\pnum
+\throws
+\tcode{bad_alloc} if \tcode{n > capacity()} is \tcode{true}.
+\end{itemdescr}
+
+\indexlibrarymember{shrink_to_fit}{inplace_vector}%
+\begin{itemdecl}
+static constexpr void shrink_to_fit() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+None.
+\end{itemdescr}
+
+\indexlibrarymember{erase}{inplace_vector}%
+\indexlibrarymember{pop_back}{inplace_vector}%
+\begin{itemdecl}
+constexpr iterator erase(const_iterator position);
+constexpr iterator erase(const_iterator first, const_iterator last);
+constexpr void pop_back();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Invalidates iterators and references at or after the point of the erase.
+
+\pnum
+\throws
+Nothing unless an exception is thrown by
+the assignment operator or move assignment operator of \tcode{T}.
```

Comment:
"The move assignment operator," said Ford Prefect quietly, "is also an assignment operator."

If this is trying to hint that we might use an assignment operator other than the move-assignment operator (e.g. if the only `operator=` available is `void operator=(std::any)`, [like this](https://godbolt.org/z/TE63YPoEz)) — well, that's true, but I still think there's no sense in mentioning a subset of assignment operators (the move assignment operators) here.

```suggestion
the assignment operator of \tcode{T}.
```

Cf. the Complexity element, which gets it right, in my book.

Reply:
this is a normative change, please forward to LWG.
resolving here to reduce "noise".

Reply:
It should either be "the copy assignment operator or move assignment operator" or "**an** assignment operator", but the approved wording and the suggested change are both bad.

Reply:
See also #3208. It was very unfortunate that the bad wording got copied.

I think "the copy assignment operator or move assignment operator" is still defective, because `T` can be copy/move-assigned by an assignment operator which is a template when the real copy/move assignment operator is not selected.

Reply:
@jwakely Could we turn this, and #3208, into an LWG issue?

Reply:
https://cplusplus.github.io/LWG/issue4123

Reply:
Thanks!